pkg/controller/publishingstrategy/publishingstrategy_controller_test.go

```diff
@@ -378,21 +396,75 @@ func TestDeleteIngressWithAnnotation(t *testing.T) {
 		t.Fatalf("Unable to add operatorv1 scheme (%v)", err)
 	}
 
-	err := r.client.Create(ctx, mockNonDefaultIngressController())
+	if err := apis.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add route scheme (%v)", err)
+	}
+
+	err := r.client.Create(ctx, mockNonDefaultIngressNoAnnotation())
+	if err != nil {
+		t.Errorf("couldn't create ingress without annotation")
+	}
+
+	err = r.client.Create(ctx, mockNonDefaultIngressController())
 	if err != nil {
 		t.Errorf("couldn't create ingresscontroller %s", err)
 	}
 
-	list := &operatorv1.IngressControllerList{}
+	err = r.client.Create(ctx, mockPublishingStrategy())
+	if err != nil {
+		t.Errorf("couldn't create ingresscontroller %s", err)
+	}
+
+	ingressControllerList := &operatorv1.IngressControllerList{}
 	opts := client.ListOptions{}
 
-	err = r.client.List(ctx, list, &opts)
+	err = r.client.List(ctx, ingressControllerList, &opts)
 	if err != nil {
 		t.Errorf("couldn't get ingresscontroller list %s", err)
 	}
 
-	err = r.deleteIngressWithAnnotation(list)
+	err = r.deleteIngressWithAnnotation(mockPublishingStrategy().Spec.ApplicationIngress, ingressControllerList)
+	if err != nil {
+		t.Fatalf("couldn't delete ingress")
+	}
+
+	// if ingress without annotation hit method, then it should not be removed
+	err = r.deleteIngressWithAnnotation(mockPublishingStrategy().Spec.ApplicationIngress, ingressControllerList)
+	if err != nil {
+		t.Fatalf("couldn't delete ingress")
+	}
+
+	if ingressControllerList.Items[0].Spec.Domain != "apps2.exaple-nondefault-domain-with-no-annotation" {
+		t.Fatalf("expect nondefault ingress to be on cluster but it is not")
+	}
+}
```

Comment:
Will `ingressControllerList` be updated by the `Delete` in the `deleteIngressWithAnnotation` method, or do you need to do an explicit `Get` from the (fake) Kubernetes client to check for the no annotation exists?